extensions/data-transfer/portability-data-transfer-imgur/src/main/java/org/datatransferproject/datatransfer/imgur/photos/ImgurPhotosImporter.java

```diff
@@ -0,0 +1,194 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+
+import java.util.Base64;
+import java.util.Map;
+import java.util.UUID;
+
+import static java.lang.String.format;
+
+/** Imports albums and photos to Imgur */
+public class ImgurPhotosImporter
+    implements Importer<TokensAndUrlAuthData, PhotosContainerResource> {
+
+  private final OkHttpClient client;
+  private final ObjectMapper objectMapper;
+  private final JobStore jobStore;
+  private final Monitor monitor;
+  private static final String BASE_URL = ImgurTransferExtension.BASE_URL;
+  private static final String CREATE_ALBUM_URL = BASE_URL + "/album";
+  private static final String UPLOAD_PHOTO_URL = BASE_URL + "/image";
+  private static final String TEMP_PHOTOS_KEY = "tempPhotosData";
+
+  public ImgurPhotosImporter(
+      Monitor monitor, OkHttpClient client, ObjectMapper objectMapper, JobStore jobStore) {
+    this.client = client;
+    this.objectMapper = objectMapper;
+    this.jobStore = jobStore;
+    this.monitor = monitor;
+  }
+
+  @Override
+  public ImportResult importItem(
+      UUID jobId, TokensAndUrlAuthData authData, PhotosContainerResource resource)
+      throws Exception {
+    if (resource == null) {
+      // Nothing to import
+      return ImportResult.OK;
+    }
+
+    // TODO: store objects containing individual mappings instead of single object containing all
+    // mappings
+    TempPhotosData photoData = jobStore.findData(jobId, createCacheKey(), TempPhotosData.class);
+
+    if (photoData == null) {
+      photoData = new TempPhotosData(jobId);
+      jobStore.create(jobId, createCacheKey(), photoData);
+    }
+
+    try {
+      // Import albums
+      for (PhotoAlbum album : resource.getAlbums()) {
+        importAlbum(album, jobId, authData, photoData);
+      }
+      // Import photos
+      for (PhotoModel photo : resource.getPhotos()) {
+        importPhoto(photo, jobId, authData);
+      }
+    } catch (IOException e) {
+      monitor.severe(() -> "Error importing item", e);
+      return new ImportResult(e);
+    }
+    return new ImportResult(ImportResult.ResultType.OK);
+  }
+
+  private void importAlbum(
+      PhotoAlbum album, UUID jobId, TokensAndUrlAuthData authData, TempPhotosData photoData)
+      throws IOException {
+    String description = album.getDescription();
+
+    Request.Builder requestBuilder = new Request.Builder().url(CREATE_ALBUM_URL);
+    requestBuilder.header("Authorization", "Bearer " + authData.getAccessToken());
+
+    FormBody.Builder builder = new FormBody.Builder().add("title", album.getName());
+    if (!Strings.isNullOrEmpty(description)) {
+      builder.add("description", description);
+    }
+    RequestBody formBody = builder.build();
+    requestBuilder.post(formBody);
+
+    Response response = client.newCall(requestBuilder.build()).execute();
+    int code = response.code();
+    if (code >= 200 && code <= 299) {
```

Comment:
Nit: can we wrap this check in a precondition instead of the if else block? 

Same with the check of body == null and Strings.isNullOrEmpty(newAlbumId)

Reply:
Done